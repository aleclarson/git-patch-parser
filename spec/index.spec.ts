import * as fs from "fs";
import * as path from "path";
import { parsePatch, parseMultiPatch } from "..";

beforeAll(() => {
  process.chdir(__dirname);
});

it("Parses file renames", () => {
  const contents = getFixture("rename.patch");
  const parsed = parsePatch(contents);
  expect(parsed).toMatchSnapshot();
});

it("Parses file deletions", () => {
  const contents = getFixture("delete.patch");
  const parsed = parsePatch(contents);
  expect(parsed).toMatchSnapshot();
});

it("Parses file additions", () => {
  const contents = getFixture("add.patch");
  const parsed = parsePatch(contents);
  expect(parsed).toMatchSnapshot();
});

it.todo("Parses trailing newlines");

describe("Reading single patch", () => {
  it("Parses the example patch", () => {
    const contents = getFixture("react.single.patch");
    const parsed = parsePatch(contents);
    expect(parsed).toMatchSnapshot();
  });
});

describe("Reading multi patch", () => {
  it("Parses the example patch", () => {
    const contents = getFixture("react.multi.patch");
    const parsed = parseMultiPatch(contents);
    expect(parsed).toMatchSnapshot();
  });
});

function getFixture(fixture: string) {
  return fs.readFileSync(path.join("__fixtures__", fixture), "utf8");
}
