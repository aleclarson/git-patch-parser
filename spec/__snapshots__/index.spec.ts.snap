// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parses file additions 1`] = `
Object {
  "changes": Array [
    Object {
      "file": ".babelrc",
      "text": "{
  \\"presets\\": [\\"@babel/env\\", \\"@babel/typescript\\"]
}
- 
.19.2

",
      "type": "add",
    },
  ],
  "message": "boop",
  "sha": "da991f76c33de0a0e3b4c824a67496d01e01ab1e",
}
`;

exports[`Parses file deletions 1`] = `
Object {
  "changes": Array [
    Object {
      "file": ".prettierrc",
      "text": "{
  \\"bracketSpacing\\": true,
  \\"jsxBracketSameLine\\": true,
  \\"singleQuote\\": false,
  \\"semi\\": true,
  \\"trailingComma\\": \\"es5\\",
  \\"arrowParens\\": \\"always\\"
}
- 
.19.2

",
      "type": "delete",
    },
  ],
  "message": "tmp",
  "sha": "e6e150aea92b5eb626b611a54505fcfb74e1a7d8",
}
`;

exports[`Parses file renames 1`] = `
Object {
  "changes": Array [
    Object {
      "file": null,
      "text": "",
      "type": "add",
    },
  ],
  "message": "tmp",
  "sha": "8046002a8c1b736cde7090f6af14e06420a5729d",
}
`;

exports[`Reading multi patch Parses the example patch 1`] = `
Array [
  Object {
    "changes": Array [
      Object {
        "file": ".meteor/.finished-upgraders",
        "text": "# This file contains information which helps Meteor properly upgrade your
# app when you run 'meteor update'. You should check it into version control
# with your project.

notices-for-0.9.0
notices-for-0.9.1
0.9.4-platform-file
notices-for-facebook-graph-api-2
",
        "type": "add",
      },
      Object {
        "file": ".meteor/.gitignore",
        "text": "local
",
        "type": "add",
      },
      Object {
        "file": ".meteor/.id",
        "text": "# This file contains a token that is unique to your project.
# Check it into your repository along with the rest of this directory.
# It can be used for purposes such as:
#   - ensuring you don't accidentally deploy one app on top of another
#   - providing package authors with aggregated statistics

ucqipf8prm38bs7dxj
",
        "type": "add",
      },
      Object {
        "file": ".meteor/packages",
        "text": "# Meteor packages used by this project, one per line.
# Check this file (and the other files in this directory) into your repository.
#
# 'meteor add' and 'meteor remove' will edit this file for you,
# but you can also edit it by hand.

meteor-platform
autopublish
insecure
",
        "type": "add",
      },
      Object {
        "file": ".meteor/platforms",
        "text": "server
browser
",
        "type": "add",
      },
      Object {
        "file": ".meteor/release",
        "text": "METEOR@1.1.0.2
",
        "type": "add",
      },
      Object {
        "file": ".meteor/versions",
        "text": "autopublish@1.0.3
autoupdate@1.2.1
base64@1.0.3
binary-heap@1.0.3
blaze@2.1.2
blaze-tools@1.0.3
boilerplate-generator@1.0.3
callback-hook@1.0.3
check@1.0.5
ddp@1.1.0
deps@1.0.7
ejson@1.0.6
fastclick@1.0.3
geojson-utils@1.0.3
html-tools@1.0.4
htmljs@1.0.4
http@1.1.0
id-map@1.0.3
insecure@1.0.3
jquery@1.11.3_2
json@1.0.3
launch-screen@1.0.2
livedata@1.0.13
logging@1.0.7
meteor@1.1.6
meteor-platform@1.2.2
minifiers@1.1.5
minimongo@1.0.8
mobile-status-bar@1.0.3
mongo@1.1.0
observe-sequence@1.0.6
ordered-dict@1.0.3
random@1.0.3
reactive-dict@1.1.0
reactive-var@1.0.5
reload@1.1.3
retry@1.0.3
routepolicy@1.0.5
session@1.1.0
spacebars@1.0.6
spacebars-compiler@1.0.6
templating@1.1.1
tracker@1.0.7
ui@1.0.6
underscore@1.0.3
url@1.0.4
webapp@1.2.0
webapp-hashing@1.0.3
",
        "type": "add",
      },
      Object {
        "file": "simple-todos-react.css",
        "text": "/* CSS declarations go here */
",
        "type": "add",
      },
      Object {
        "file": "simple-todos-react.html",
        "text": "<head>
  <title>simple-todos-react</title>
</head>

<body>
  <h1>Welcome to Meteor!</h1>

  {{> hello}}
</body>

<template name=\\"hello\\">
  <button>Click Me</button>
  <p>You've pressed the button {{counter}} times.</p>
</template>
",
        "type": "add",
      },
      Object {
        "file": "simple-todos-react.js",
        "text": "if (Meteor.isClient) {
  // counter starts at 0
  Session.setDefault('counter', 0);

  Template.hello.helpers({
    counter: function () {
      return Session.get('counter');
    }
  });

  Template.hello.events({
    'click button': function () {
      // increment the counter when button is clicked
      Session.set('counter', Session.get('counter') + 1);
    }
  });
}

if (Meteor.isServer) {
  Meteor.startup(function () {
    // code to run on server at startup
  });
}
-
.2.1


",
        "type": "add",
      },
    ],
    "message": "Step 1: Run \`meteor create\`",
    "sha": "d7a5e7f791c6f6659c4767f8bb4afa853d793647",
  },
  Object {
    "changes": Array [
      Object {
        "diff": Array [
          Object {
            "inputRange": Object {
              "length": 3,
              "start": 7,
            },
            "lines": Array [
              Object {
                "prefix": " ",
                "text": "meteor-platform",
              },
              Object {
                "prefix": " ",
                "text": "autopublish",
              },
              Object {
                "prefix": " ",
                "text": "insecure",
              },
              Object {
                "prefix": "+",
                "text": "react",
              },
            ],
            "outputRange": Object {
              "length": 4,
              "start": 7,
            },
          },
        ],
        "file": ".meteor/packages",
        "type": "change",
      },
      Object {
        "diff": Array [
          Object {
            "inputRange": Object {
              "length": 5,
              "start": 1,
            },
            "lines": Array [
              Object {
                "prefix": " ",
                "text": "autopublish@1.0.3",
              },
              Object {
                "prefix": " ",
                "text": "autoupdate@1.2.1",
              },
              Object {
                "prefix": "+",
                "text": "babel-compiler@5.6.15",
              },
              Object {
                "prefix": "+",
                "text": "babel-runtime@0.1.1",
              },
              Object {
                "prefix": " ",
                "text": "base64@1.0.3",
              },
              Object {
                "prefix": " ",
                "text": "binary-heap@1.0.3",
              },
              Object {
                "prefix": " ",
                "text": "blaze@2.1.2",
              },
            ],
            "outputRange": Object {
              "length": 7,
              "start": 1,
            },
          },
        ],
        "file": ".meteor/versions",
        "type": "change",
      },
    ],
    "message": "Step 2.1: Add the React package",
    "sha": "96bd5130dbd707aa497a0effa85cb7316cd7d560",
  },
]
`;

exports[`Reading single patch Parses the example patch 1`] = `
Object {
  "changes": Array [
    Object {
      "diff": Array [
        Object {
          "inputRange": Object {
            "length": 15,
            "start": 1,
          },
          "lines": Array [
            Object {
              "prefix": " ",
              "text": "// App component - represents the whole app",
            },
            Object {
              "prefix": " ",
              "text": "App = React.createClass({",
            },
            Object {
              "prefix": "-",
              "text": "  getTasks() {",
            },
            Object {
              "prefix": "-",
              "text": "    return [",
            },
            Object {
              "prefix": "-",
              "text": "      { _id: 1, text: \\"This is task 1\\" },",
            },
            Object {
              "prefix": "-",
              "text": "      { _id: 2, text: \\"This is task 2\\" },",
            },
            Object {
              "prefix": "-",
              "text": "      { _id: 3, text: \\"This is task 3\\" }",
            },
            Object {
              "prefix": "-",
              "text": "    ];",
            },
            Object {
              "prefix": "+",
              "text": "",
            },
            Object {
              "prefix": "+",
              "text": "  // This mixin makes the getMeteorData method work",
            },
            Object {
              "prefix": "+",
              "text": "  mixins: [ReactMeteorData],",
            },
            Object {
              "prefix": "+",
              "text": "",
            },
            Object {
              "prefix": "+",
              "text": "  // Loads items from the Tasks collection and puts them on this.data.tasks",
            },
            Object {
              "prefix": "+",
              "text": "  getMeteorData() {",
            },
            Object {
              "prefix": "+",
              "text": "    return {",
            },
            Object {
              "prefix": "+",
              "text": "      tasks: Tasks.find({}).fetch()",
            },
            Object {
              "prefix": "+",
              "text": "    }",
            },
            Object {
              "prefix": " ",
              "text": "  },",
            },
            Object {
              "prefix": " ",
              "text": "",
            },
            Object {
              "prefix": " ",
              "text": "  renderTasks() {",
            },
            Object {
              "prefix": "-",
              "text": "    return this.getTasks().map((task) => {",
            },
            Object {
              "prefix": "+",
              "text": "    // Get tasks from this.data.tasks",
            },
            Object {
              "prefix": "+",
              "text": "    return this.data.tasks.map((task) => {",
            },
            Object {
              "prefix": " ",
              "text": "      return <Task key={task._id} task={task} />;",
            },
            Object {
              "prefix": " ",
              "text": "    });",
            },
            Object {
              "prefix": " ",
              "text": "  },",
            },
            Object {
              "prefix": "-",
              "text": "- ",
            },
          ],
          "outputRange": Object {
            "length": 19,
            "start": 1,
          },
        },
      ],
      "file": "App.jsx",
      "type": "change",
    },
  ],
  "message": "Step 3.2: Modify App component to get tasks from
 collection",
  "sha": "ce86ff010f94a8a1cc2c8e6f9331df080565682b",
}
`;
